core/consensus/grandpa/impl/chain_impl.cpp

```diff
@@ -0,0 +1,85 @@
+/**
+ * Copyright Soramitsu Co., Ltd. All Rights Reserved.
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include "consensus/grandpa/impl/chain_impl.hpp"
+
+#include <utility>
+
+#include <boost/optional/optional_io.hpp>
+
+OUTCOME_CPP_DEFINE_CATEGORY(kagome::consensus::grandpa, ChainImpl::Error, e) {
+  using E = kagome::consensus::grandpa::ChainImpl::Error;
+  switch (e) {
+    case E::BLOCK_AFTER_LIMIT:
+      return "target block is after limit";
+  }
+  return "Unknown error";
+}
+
+namespace kagome::consensus::grandpa {
+
+  using primitives::BlockHash;
+  using primitives::BlockNumber;
+
+  ChainImpl::ChainImpl(
+      std::shared_ptr<blockchain::BlockTree> block_tree,
+      std::shared_ptr<blockchain::BlockHeaderRepository> header_repository,
+      common::Logger logger)
```

Comment:
make logger as a private member (and init it in place). 
it is a requirement for injector